src/libopensc/card-itacns.c

```diff
@@ -101,23 +104,30 @@ static int itacns_match_cns_card(sc_card_t *card, unsigned int i)
 
 	/* H13 */
 	/* Version byte: h.l, h in the high nibble, l in the low nibble. */
-	if(card->driver) {
-		DRVDATA(card)->cns_version = atr[i];
-	}
 	/* Warn if version is not 1.X. */
 	if(atr[i] != 0x10 && atr[i] != 0x11) {
 		char version[8];
 		snprintf(version, sizeof(version), "%d.%d", (atr[i] >> 4) & 0x0f, atr[i] & 0x0f);
 		sc_log(card->ctx, "CNS card version %s; no official specifications "
 			"are published. Proceeding anyway.\n", version);
 	}
+	card->version.hw_major = (atr[i] >> 4) & 0x0f;
+	card->version.hw_minor = atr[i] & 0x0f;
 	i++;
 
 	itacns_atr_match(i, 0x31); /* H14 */
 	itacns_atr_match(i, 0x80); /* H15 */
 
 	card->type = SC_CARD_TYPE_ITACNS_CNS;
 
+	if (card->driver) {
+		if (DRVDATA(card)->ic_manufacturer_code == ITACNS_ICMAN_INFINEON &&
```

Comment:
in `match_card` don't add or change the internal data structures. Please use init_card for this. (Yes, I see that this is already present in the current code base, but since you have your hands on this code, please move this part to the card's initialization, thank you)

Reply:
Done as per comment before